App.js

```diff
@@ -1,9 +1,363 @@
 import React, { useState, useCallback } from 'react';
-import { StyleSheet, Text, View, FlatList, Dimensions, TouchableOpacity} from 'react-native';
+import { StyleSheet, Text, View, FlatList, Dimensions, TouchableHighlight } from 'react-native';
 import { newBoard } from './newboard.js';
 
 export default function App() {
-  
+  const [{ roll, even, odd, position, message, optMessage, score, board }, setGameState] = useState({
+    roll: 0,
+    even: false,
+    odd: false,
+    position: 0,
+    message: 'Roll again',
+    optMessage: 'Kim',
+    score: 0,
+    board: JSON.parse(JSON.stringify(newBoard)),
+  });
+
+  // Nav bar
+  const nav = [
+    { key: 1, name: 'PLAY', invisible: false },
+    { key: 2, name: '', invisible: true },
+    { key: 3, name: '', invisible: true },
+    { key: 4, name: '', invisible: true },
+    { key: 5, name: '', invisible: true },
+    { key: 6, name: 'Score' },
+    { key: 7, name: 0 },
+  ];
+
+  const numColumns = 7;
+
+  // render Nav Bar
+  const renderNav = ({ item }) => {
+    if (item.invisible === true) {
+      return <View style={[styles.item, styles.itemInvisible]} />;
+    }
+    if (item.name === 'PLAY') {
+      return (
+        <View style={styles.item}>
+          <TouchableHighlight onPress={() => pressPlay()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    if (item.key === 7) {
```

Comment:
can you just give item `7` a name? so you can do like item.name === 'Score'

Reply:
I render all names onto the NAV or BOARD. If I added a name here...then I would have change my render code to ignore this one. How about instead I make a constant that describes what this is?

const scoreNumberItem = 7;

if (item.key === scoreNumberItem) {